src/Shrubbery/Classes.hs

```diff
@@ -88,10 +89,11 @@ class Unification a where
   in 'BranchTypes a'. If there are duplicate types in 'BranchTypes a', you
   should use 'unifyWithIndex' to disambiguate them.
 -}
-unify :: ( KnownNat branchIndex
-         , branchIndex ~ FirstIndexOf t (BranchTypes a)
-         , Unification a
-         )
+unify :: forall t a branchIndex.
```

Comment:
I added this 'forall' so that 't' would appear first for the purpose of allowing nice 'TypeApplication' usage.